addons/format_lame.c

```diff
@@ -0,0 +1,566 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * Copyright (C) 2024, Naveen Albert, sponsored by Marfox Ltd.
+ * Copyright (C) 2024, Sperl Viktor
+ * 
+ * Naveen Albert <asterisk@phreaknet.org>
+ * Sperl Viktor <viktike32@gmail.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+/*!
+ * \file
+ * \brief MP3 Format including encoder and decoder
+ * 
+ * \author Naveen Albert <asterisk@phreaknet.org>
+ * \author Sperl Viktor <viktike32@gmail.com>
+ * 
+ * \ingroup formats
+ */
+
+/*** MODULEINFO
+	<depend>lame</depend>
+	<conflict>format_mp3</conflict>
+	<defaultenabled>no</defaultenabled>
+	<support_level>extended</support_level>
+ ***/
+
+#include "asterisk.h"
+
+#include <stdarg.h>
+
+#include <lame/lame.h>
+
+#include "asterisk/module.h"
+#include "asterisk/mod_format.h"
+#include "asterisk/logger.h"
+#include "asterisk/format_cache.h"
+
+#define SLIN_BUFLEN 320
+#define DECORDER_OUTLEN 32768
+#define BITS 2
+
+struct mp3lame_private {
+	lame_global_flags * lgfp;
+	hip_global_flags * hgfp;
+
+	int sample_rate;
+
+	/* decoder buffer */
+	short int doutput[DECORDER_OUTLEN *  1024];
+	long doffset;
+	int dsamples;
+
+	/* decoder positions */
+	int total_bytes_decoded, total_bytes_compressed;
+
+	/* counters */
+	unsigned int decoder_counter, encoder_counter;
+};
+
+static void mp3lame_error(const char *format, va_list ap)
+{
+	ast_log(LOG_WARNING, format, ap);
```

Comment:
Same error, the compiler doesn't like being passed a `const char*` instead of a literal...
`format_lame.c:73:9: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]`

Reply:
Okay, I see now. This was the previous solution: https://github.com/asterisk/asterisk/pull/665/files#diff-2d5e3732e5f2298375567306889cfed1863e7b30ce577000af4b0fb56b814861R390
However I've removed these logging functions as I never saw any calls to these anyway.

Reply:
> Okay, I see now. This was the previous solution: https://github.com/asterisk/asterisk/pull/665/files#diff-2d5e3732e5f2298375567306889cfed1863e7b30ce577000af4b0fb56b814861R390 However I've removed these logging functions as I never saw any calls to these anyway.

Those functions are used, here: https://github.com/asterisk/asterisk/pull/665/files#diff-2d5e3732e5f2298375567306889cfed1863e7b30ce577000af4b0fb56b814861R415

I personally would like them included since otherwise if an error occurs, nothing will be sent to the logger.

They are the logging callbacks for LAME. I don't think that was the error I recall seeing when I was working on this, but it's been long enough that I don't quite remember. It did compile fine on Debian.